apps/lib/apps.c

```diff
@@ -2518,46 +2531,21 @@ BIO *bio_open_owner(const char *filename, int format, int private)
     return NULL;
 }
 
-static BIO *bio_open_default_(const char *filename, char mode, int format,
-                              int quiet)
+BIO *bio_open_default(const char *filename, char mode, int format)
 {
     BIO *ret;
 
-    if (filename == NULL || strcmp(filename, "-") == 0) {
-        ret = mode == 'r' ? dup_bio_in(format) : dup_bio_out(format);
-        if (quiet) {
-            ERR_clear_error();
-            return ret;
-        }
-        if (ret != NULL)
-            return ret;
-        BIO_printf(bio_err,
-                   "Can't open %s, %s\n",
-                   mode == 'r' ? "stdin" : "stdout", strerror(errno));
-    } else {
-        ret = BIO_new_file(filename, modestr(mode, format));
-        if (quiet) {
-            ERR_clear_error();
-            return ret;
-        }
-        if (ret != NULL)
-            return ret;
-        BIO_printf(bio_err,
-                   "Can't open %s for %s, %s\n",
+    /* Using stdin or stdout? */
+    if (filename == NULL || strcmp(filename, "-") == 0)
+        return mode == 'r' ? dup_bio_in(format) : dup_bio_out(format);
+
+    ret = BIO_new_file(filename, modestr(mode, format));
+    if (ret == NULL) {
```

Comment:
a job for your new check function?

Reply:
Okay, but |mode| doesn't quite fit, going to leave it as 'r' if that's okay.